backend/wmg/api/v1.py

```diff
@@ -1,58 +1,157 @@
+        cube = tiledb.open(cube_uri, ctx=create_ctx(fast_config()))
 
-    def build_viz_dots(cell_type_term_ids_: List[str]) -> List[dict]:
-        return [
-            build_viz_dot(cell_type_term_id, me=0.0, pc=0.0, n=0, tpc=0.0) for cell_type_term_id in cell_type_term_ids_
-        ]
+    return cube
 
-    def build_tissues_types(tissue_type_term_ids_: List[str]) -> Dict[str, List]:
-        return dict(
-            [(tissue_type_term_id, build_viz_dots(cell_type_term_ids)) for tissue_type_term_id in tissue_type_term_ids_]
-        )
 
-    def build_genes(gene_term_ids_: List[str]) -> Dict[str, dict]:
-        return dict([(gene_term_id, build_tissues_types(tissue_type_term_ids)) for gene_term_id in gene_term_ids_])
+def build_cube_uri():
+    # TODO: Retrieve from app config
+    cube_base_uri = "s3://wmg-dev"
+    # TODO: Retrieve from s3://wmg-<env>/latest_snapshot_uuid
+    cube_latest_snapshot = "dummy-snapshot"
+    cube_uri = f"{cube_base_uri}/{cube_latest_snapshot}/cube/"
+    return cube_uri
+
 
-    def build_gene_id_label_mapping(gene_term_ids_) -> List[dict]:
-        return [{gene_term_id: f"{gene_term_id}_label"} for gene_term_id in gene_term_ids_]
+def query():
+    request = connexion.request.json
+
+    criteria = WmgQueryCriteria(**request["filter"])
+    query_result = WmgQuery(find_cube_latest_snapshot()).execute(criteria)
+    dot_plot_matrix_df = build_dot_plot_matrix(query_result)
+    all_filter_dims_values = extract_filter_dims_values(query_result)
+
+    include_filter_dims = request.get("include_filter_dims", False)
+    response_filter_dims_values = build_filter_dims_values(all_filter_dims_values) if include_filter_dims else {}
 
-    def build_cell_type_id_label_mapping(cell_type_term_ids_) -> List[dict]:
-        return [{cell_type_term_id: f"{cell_type_term_id}_label"} for cell_type_term_id in cell_type_term_ids_]
+    cell_type_term_ids = all_filter_dims_values["cell_type_ontology_term_id"]
 
     return jsonify(
         dict(
             snapshot_id=DUMMY_SNAPSHOT_UUID,
-            expression_summary=build_genes(gene_term_ids),
+            expression_summary=build_expression_summary(dot_plot_matrix_df),
             term_id_labels=dict(
-                genes=build_gene_id_label_mapping(gene_term_ids),
-                cell_types=build_cell_type_id_label_mapping(cell_type_term_ids),
+                genes=build_gene_id_label_mapping(criteria.gene_ontology_term_ids),
+                cell_types=build_ontology_term_id_label_mapping(cell_type_term_ids),
             ),
+            filter_dims=response_filter_dims_values,
+        )
+    )
+
+
+# TODO: It would be reasonable to fetch datasets via a REST API call to Data Portal, rather than coupling to the
+#  portal's db layer, considering that this is the only occasion where the backend.wmg.api package introduces a
+#  dependency on the db. There is no appropriate API call at this time and the lower of making an API call would
```

Comment:
I think you might be missing a word here? Unless there is a massive issue with performance I think it makes more sense to create an endpoint in the data portal to get this information for a given list of datasets... or add it to the cube

Reply:
ah, yes: `and the lower [performance] of making` will fix

Reply:
Good point regarding writing this out at generation-time. It's in keeping with "snapshotted" data and consistency across all the data sources.  I think we'd have to write out as a new data artifact, rather than putting that data into the cube (it would be highly repetitive data, i.e. denormalized).   And this provides even less coupling between WMG and Data Portal than even just making an API call.

Reply:
https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/2085
https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/2086